src/test/memory/test_mmap.rs

```diff
@@ -0,0 +1,354 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+use test_utils::set;
+use test_utils::TestEnvironment as TestEnv;
+
+const MAPLEN: usize = 16;
+
+fn validate_shadow_access(buf: &[u8]) -> Result<(), String> {
```

Comment:
Optional throughout - if you use `Box<dyn Error>` as the error types, you won't need all the `map_err` calls. For example, see https://github.com/shadow/shadow/blob/dev/src/main/utility/proc_maps.rs#L185